cpu/cc26x0/periph/gpio.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 2016 Leon George
+ * (who won't go after you should you copy any part of this file ;-)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+/**
+ * @ingroup     driver_periph
+ * @{
+ *
+ * @file
+ * @brief       low-level GPIO driver implementation
+ *
+ * @author      Leon M. George <leon@georgemail.eu>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "sched.h"
+#include "thread.h"
+#include "periph/gpio.h"
+#include "periph_conf.h"
+
+#include "cc26x0-ioc.h"
+#include "cc26x0-gpio.h"
+#include "cc26x0-prcm.h"
+
+#define GPIO_ISR_CHAN_NUMOF             (32)
+
+typedef struct {
+    void (*cb)(void *arg);
+    void *arg;
+} gpio_ctx_t;
+
+static gpio_ctx_t gpio_chan[GPIO_ISR_CHAN_NUMOF];
+
+int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
+{
+    if ((pin < 0) || (pin > 31))
+        return -1;
+
+    if (!(PRCM->PDSTAT0 & PDSTAT0_PERIPH_ON)) {
```

Comment:
no need to check, just override the value of the reg, saves some instructions...